From 46aba8f887616f8a8ab136327c1fa88fc1f704f9 Mon Sep 17 00:00:00 2001
From: hariharanr <hariharanr@ami.com>
Date: Fri, 26 Apr 2024 14:28:12 +0530
Subject: [PATCH 7/8] OT-3805 AST2700 Support

AST2700 uses different vhub. This commit enables support for UDC
configuration.

Signed-off-by: Mohammed Javith Akthar M <mohammedjavitham@ami.com>
Signed-off-by: hariharanr <hariharanr@ami.com>

%% original patch: 0007-OT-3805-AST2700-Support.patch
---
 src/system.hpp | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/src/system.hpp b/src/system.hpp
index 8b8636e..9417a05 100644
--- a/src/system.hpp
+++ b/src/system.hpp
@@ -744,6 +744,10 @@ struct UsbGadget
         const fs::path configDir = gadgetDir / "configs/c.1";
         const fs::path massStorageDir = configDir / "mass_storage.usb0";
         const fs::path configStringsDir = configDir / "strings/0x409";
+        const std::string usbVirtualHub =
+            fs::exists("/sys/bus/platform/devices/12011000.usb-vhub")
+                ? "12011000" /* AST2700 */
+                : "1e6a0000";
 
         /* Parameters for Session management register/unregister */
 
@@ -869,12 +873,14 @@ struct UsbGadget
                     }
                 }
 
-                for (const auto& port : fs::directory_iterator(
-                         "/sys/bus/platform/devices/1e6a0000.usb-vhub"))
+                for (const auto& port :
+                     fs::directory_iterator("/sys/bus/platform/devices/" +
+                                            usbVirtualHub + ".usb-vhub"))
                 {
                     const std::string portId = port.path().filename();
 
-                    if (portId.find("1e6a0000.usb-vhub:p") != std::string::npos)
+                    if (portId.find(usbVirtualHub + ".usb-vhub:p") !=
+                        std::string::npos)
                     {
                         constexpr std::string_view portDelimiter = ":p";
                         const std::string portNumber = portId.substr(
-- 
2.34.1

